update/Pos/Binary/Update.hs

```diff
@@ -74,23 +73,32 @@ instance Bi U.BlockVersionData where
         put bvdUpdateSoftforkThd
 
 instance Bi U.UpdateProposal where
-    get = label "UpdateProposal" $
-          U.UpdateProposal
-            <$> get
-            <*> get
-            <*> get
-            <*> getUpData
-            <*> get
-      where getUpData = do   -- Check if proposal data is non-empty
-                pd <- get
-                when (HM.null pd) $
-                    fail "Pos.Binary.Update: UpdateProposal: empty proposal data"
-                return pd
-    put U.UpdateProposal {..} =  put upBlockVersion
+    get = label "UpdateProposal" $ do
+        d <- get
```

Comment:
lol, no. I have invited new error-free deserialization, because your deserialization is correct if you can read word.